ncigrafana/parse_membership.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+
+"""
+Copyright 2020 ARC Centre of Excellence for Climate Systems Science
+
+author: Aidan Heerdegen <aidan.heerdegen@anu.edu.au>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from __future__ import print_function
+
+import argparse
+import datetime
+import sys
+from .UsageDataset import *
+from .DBcommon import extract_num_unit, parse_size, mkdir, archive
+from .DBcommon import datetoyearquarter, date_range_from_quarter
+
+dbfileprefix = '.'
+nfields = 7
+
+def parse_membership(filename, verbose, db=None, dburl=None):
+
+    project = None
+    date = None
+    system = 'gadi'
+
+    with open(filename) as f:
+
+        print("Parsing {file}".format(file=filename))
+
+        # read 3 lines in membership dump file
+        header = f.readline()
+        if verbose: print("> ", header)
+        
+        date_line = f.readline()
+        if verbose: print("> ", header)
+
+        data_line = f.readline()
+
+        # parse date from file
+        date = datetime.datetime.strptime(date_line.strip(), '%a %b %d %H:%M:%S %Z %Y')
+
+        # parse project_id and members from file
+        data_line_parts = data_line.strip().split(":")
+        project = data_line_parts[0]
+        members_csv = data_line_parts[3]
+        members_list = members_csv.split(",")
+
+        # remove service users
+        members_list = [ m for m in members_list if not m.endswith("_thredds") ]
+        members_list = [ m for m in members_list if not m.endswith("_ci") ]
```

Comment:
Note that other service users might not follow the pattern `*_ci`

Reply:
`ki32` is a good project to check out the possible spectrum of service users, and lordy!, there is no pattern at all
```
accesstester 
hc46_gitlab 
ki32_nfs
tm70_ci
wp25_ci 
yj2_repoweb
```

I looked in the LDAP record, and the only thing I could find that was different about the service users is they don't have either the `mail:` field or `labeledURI` field.

My suggestion would be to add a column in the users table to indicate service users and just manually enter the ones we know, and filter them out of reports as required.

We can try automating that process by checking if the above fields exist, but that is problematic if they decide to change how the LDAP records are returned.

Reply:
I will keep an out for others